src/test/ui/consts/const-fn-error.stderr

```diff
@@ -40,7 +40,7 @@ error[E0080]: evaluation of constant value failed
 LL |     for i in 0..x {
    |              ^^^^
    |              |
-   |              calling non-const function `<std::ops::Range<usize> as IntoIterator>::into_iter`
+   |              calling non-const function `iter::range::<impl Iterator for std::ops::Range<usize>>::next`
```

Comment:
It seems that if a non-const function is called, unstably const functions don't get a mention. Is that how it is meant to be?

Reply:
I don't know how that test works. Somehow despite all the errors we get from checking the MIR for feature gates and non-const functions, it still gets fed into the CTFE, which emits this error.

Perhaps someone else from @rust-lang/wg-const-eval can tell what's going on?

I wouldn't call this allowing calling unstablly const fns in stable code, since there are already errors.


Once we allow `Iterator` to have `const` `impl`s, we would add a test to see if for loops are gated, right now I don't think any actions will be necessary.

Reply:
> Somehow despite all the errors we get from checking the MIR for feature gates and non-const functions, it still gets fed into the CTFE, which emits this error.

This might be something similar to https://github.com/rust-lang/rust/issues/79047, https://github.com/rust-lang/rust/issues/85907.
Some errors don't abort compilation (so that we can show more than 1 error) and we don't have the proper logic in place to skip CTFE of things that are already 'tainted' by errors.